core/src/components/input/test/item/input.e2e.ts

```diff
@@ -36,4 +36,35 @@ test.describe('input: item', () => {
     const list = page.locator('ion-list');
     expect(await list.screenshot()).toMatchSnapshot(`input-inset-list-no-fill-${page.getSnapshotSettings()}.png`);
   });
+  test('input renders as modern when label is set asynchronously', async ({ page }, testInfo) => {
+    testInfo.annotations.push({
+      type: 'issue',
+      description: 'https://github.com/ionic-team/ionic-framework/issues/27085',
+    });
+
+    await page.setContent(
+      `<ion-item>
+        <ion-input></ion-input>
+      </ion-item>
+      <ion-button onclick="setLabel()">Set label</ion-button>
+
+      <script>
+        function setLabel() {
+          const input = document.querySelector('ion-input');
+          input.label = 'New label';
+        }
+      </script>
+      `
+    );
```

Comment:
Any particular reason why you did not do the following instead?

```js
await input.evaluate((el: HTMLIonInputElement) => el.value = 'New label');
```

Reply:
I _think_ you meant `el.label`, but yes this can be simplified to use Playwright's `evaluate` method to change the label vs. having the test template have a script for it. I'll update that.

The only tests that should be using the legacy syntax are the range tests (for now). This is the same behavior as `main` today (these screenshot tests are capturing the legacy output). While migrating the range tests I ran into the clipping bug: #27199, which I decided against committing the buggy screenshots. I can split that PR out if that would help with this review.

Reply:
Yep, I meant el.label`. Good catch!